### PR TITLE
Fix Mojolicious 9.0 compatibility

### PR DESCRIPTION
**Pull request by kraih**
1. Reserved stash values.
```
Route pattern "/children/:status" contains a reserved stash value at lib/OpenQA/WebAPI.pm line 361.
```
It is forbidden to use reserved stash values as placeholders in routes.

2. New log formatting

Individual log messages are now joined with a whitespace character instead of a newline. Additionally `Mojo::Log::Colored` is incompatible with Mojolicious 9.0 (reported upstream).

3. Missing controllers are now an exception
```
[error] [1Whlu6Jt] Controller "OpenQA::WebAPI::Search" does not exist at extlib/lib/perl5/Mojolicious.pm line 124.
```
One of the routes used `->to('search#search')` as a hack to render the template `search/search.html.ep`.

4. `Mojo::IOLoop::Delay` no longer exists
```
Can't locate object method "delay" via package "Mojo::IOLoop" at t/26-controllerrunning.t line 45.
```
The module is no longer part of Mojolicious, since `Mojo::Promise` exists as a much better alternative.

Progress: https://progress.opensuse.org/issues/88609

### CODE REVIEW

**Review by Martchus — DISMISSED** [review]

File: lib/OpenQA/WebAPI/Controller/API/V1/Mm.pm [hunk 1]
```diff
@@ -52,7 +52,7 @@ JSON block with the list.
 # IMHO it should be replaced with get_children and removed
 sub get_children_status {
     my ($self) = @_;
-    my $status = $self->stash('status');
+    my $status = $self->stash('state');
```
Comment (okurz) [inline lib/OpenQA/WebAPI/Controller/API/V1/Mm.pm:55]: could you rename the variable name as well to be consistent? And how about the method name and hence the routing target?

Comment (kraih) [reply lib/OpenQA/WebAPI/Controller/API/V1/Mm.pm:55]: Once i have all the other incompatibilities fixed.

**Comment by kraih** [comment]
Side note, i'm a little concerned that `Mojo::Log::Colored` was used automatically when it was installed, but had no tests whatsoever.

**Comment by kraih** [comment]
I believe all issues are resolved now and the code is compatible with 8.73 and 9.0.

**Review by Martchus — APPROVED** [review]

**Review by okurz — CHANGES_REQUESTED** [review]
What you put into the description of the pull request could also be in the commit message details. I don't know how you created your pull request but your git commit messages has *only* the subject line while the pull request description has more details. Please keep in mind that the github pull request description is only visible on github, the commit can be considered permanent information storage.

I can recommend the tool [hub](https://github.com/github/hub) (zypper in rubygem-hub) for easier PR creation. Also, I myself use a script [pr-last](http://github.com/okurz/scripts/tree/master/pr-last) to create a PR with proper description for these simple one-commit PRs:
```bash
$ cat $(which git-pr-last )
#!/bin/sh -e
target="${target:-"$USER"}"
git push $target && git show --no-patch --format=%B | hub pull-request -F -
```

And the "unstable" test t/ui/27-plugin_obs_rsync_status_details.t is becoming really annoying. I have the suspicion that another recent dependency change made it even more unreliable. In https://github.com/os-autoinst/openQA/pull/3728 I applied further workarounds with longer timeout and more retries. After you fixed the git commit message please rebase on top.

**Review by okurz — CHANGES_REQUESTED** [review]
I think I have a better idea instead of simply removing color support: Please take a look into https://github.com/os-autoinst/os-autoinst/pull/1403/files for an approach which I think you can apply here as well unless you would prefer to fix Mojo::Log::Colored and wait for that. I would appreciate if you would withhold any breaking submission of Mojolicious 9.0 until we can resolve this.

@Martchus I see that you have approved the change without further comments. Maybe you have overlooked that kraih removed the support for colors. I am not a friend of removing features without further consideration.

**Comment by kraih** [comment]
That really wasn't a feature. It was barely a hack. Since there's no tests we should really first discuss the actual requirements for the feature. That seems rather out of scope for this bug fix. I don't mind doing it, but it should really be done as a separate PR.

**Comment by kraih** [comment]
Probably worth mentioning that `Mojo::Log` already supports journald highlighting natively.

**Comment by okurz** [comment]
> Probably worth mentioning that `Mojo::Log` already supports journald highlighting natively.

sounds good. How would we make use of that?

**Comment by kraih** [comment]
> > Probably worth mentioning that `Mojo::Log` already supports journald highlighting natively.
> 
> sounds good. How would we make use of that?

It's the "short" logging option (since journald has its own timestamps etc.). https://docs.mojolicious.org/Mojo/Log#short

**Comment by kraih** [comment]
Actually, we might just solve the `Mojo::Log` issue upstream and make colour highlighting a core mojo feature too. In addition to the journald format. Then for openQA we could just switch around with environment variables depending on use case (`MOJO_LOG_SHORT=1` or `MOJO_LOG_COLOUR=1`).

**Review by Martchus — APPROVED** [review]
I approved it because I don't care about the colors. If the dependency breaks with the latest Mojolicious version I wouldn't try to support it any longer.

**Comment by kraih** [comment]
Yes, i consider this PR finished and will just add some more details to the commit message.

**Comment by kraih** [comment]
@okurz Commit message has been updated, and log coloring will be handled upstream. I'm still interested in the specific use cases we had for that though, so the upstream changes don't end up not addressing them.

**Comment by kraih** [comment]
Colored log messages have been added upstream and should be in Mojolicious 9.01. https://github.com/mojolicious/mojo/compare/363ea5575334...0cf164f56956

**Comment by okurz** [comment]
I appreciate the details in the git commit message but I would prefer if we can make "drop of a feature" mentioned prominently in a git commit message subject line so that we can see it explicitly in the package changelogs as well. So it would be great if you can just move that specific removal part into its own git commit, then we can accept it

**Comment by Martchus** [comment]
I'd like to merge this PR rather soon. Otherwise all other PRs are blocked because the CI is not working.

**Comment by okurz** [comment]
sure, as soon as we do not hide removal of a feature behind "fix ..." :)

**Comment by kraih** [comment]
I'm afraid it will take some time for me to figure out how to cleanly separate the commits.

**Comment by Martchus** [comment]
The commit for 9.01 should work. I was going to push it as well but took the opportunity to test the command from another PR (and it apparently takes forever with all the downloading and installation involved): https://github.com/os-autoinst/openQA/pull/3733/files#r576846271

**Review by okurz — APPROVED** [review]

**Comment by okurz** [comment]
package tests are fine, we should not block other problems due to incomplete patch coverage

**Pull request merged**